src/Library/DeviceBase.cpp

```diff
@@ -90,14 +90,20 @@ namespace usbguard
      * Set connect type
      */
     int retry = 0;
-    std::string connect_type;
-    while (retry < 20) {
+    std::string connect_type = "";
+    /*
+     * Read port/connect_type only for non Host Controller devices
+     */
+    if (getPort().substr(0, 3) != "usb") {
+      while (retry < 10) {
         connect_type = sysfs_device.readAttribute("port/connect_type", /*strip_last_null=*/true, /*optional=*/true);
         if (connect_type != "") {
             break;
         }
         USBGUARD_LOG(Trace) << "connect_type is empty, will attempt again";
+        usleep(200);
         retry++;
+      }
     }
     USBGUARD_LOG(Info) << "***Setting connect type=" << connect_type <<" after retry"<< retry;
```

Comment:
I forgot: It would be great to have this output line mention how long things took, e.g. so that user can proof based on logs whenever the current values are not sufficient, in the future. Consider my earlier demo code doing something like that CC-0 ("public domain") in case you want to copy from there.